sections/Chapter 5 Sprint 3/db_design_redone.tex

```diff
@@ -0,0 +1,148 @@
+\section{Database redesign}
+
+As previously mentioned in section \ref{currentState}, the code from last year was left in an unfinished and undocumented state. We estimated that it would be faster and easier to scrap it and start over rather than spending time trying to understand and document it.
+However, the database design seemed good enough to work with despite its flaws. 
+We therefore decided to keep it as we were under time constraints.
+
+In order to communicate with the database from the C\# application, we opted to use EF Core.
+Because EF Core is designed with the Code First approach in mind, we are unable to simply make SQL queries in the code itself as the old group did. 
+Instead, we had to model the current database design using C\# classes. 
+
+This caused problems as the current design did not translate well into an object-oriented approach.
+To execute what would otherwise be considered a relative simple SQL query, we had to write a ton of code which ultimately defeated the purpose of the Code First approach. 
+The most important method for the WordCount controller, the POST method, can be seen in code snippet \ref{lst:old_controller} for the old database design.
```

Comment:
Why is it the most important? Is fetching data from the database not equally important?

Reply:
Maybe rephrase it to "the most complex" or "most comprehensive"?

Reply:
```suggestion
The most complex method for the WordCount controller, the POST method, can be seen in code snippet \ref{lst:old_controller} for the old database design.
```